codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -277,9 +265,8 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
         public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers headers, int streamDependency,
                 short weight, boolean exclusive, int padding, boolean endOfStream) throws Http2Exception {
             Http2Stream stream = connection.stream(streamId);
-            verifyGoAwayNotReceived();
-            if (shouldIgnoreFrame(stream, false)) {
-                // Ignore this frame.
+
+            if (stream != null && (stream.state() == CLOSED || streamCreatedAfterGoAwaySent(stream))) {
```

Comment:
We already have a check for `stream != null` below in the else condition....and we already check the stream state (case statement for `HALF_CLOSED_REMOTE` and `CLOSED`).  I wonder if we can consolidate the checks...by just adding `stream.state() == CLOSED || streamCreatedAfterGoAwaySent(stream))` after the `case CLOSED:`.  We can reduce the duplication this way....I don't think we lose much in clarity either.